src/Fantomas.Tests/ClassTests.fs

```diff
@@ -433,3 +433,41 @@ let ``no extra new lines between type members, 569``() =
 """
 
     formatSourceString false original config |> should equal original
+
+[<Test>]
+let ``no extra new line before nested module with attribute, 586``()=
+    let original = """module A =
```

Comment:
Maybe start using [shouldNotChangeAfterFormat](https://github.com/fsprojects/fantomas/blob/master/src/Fantomas.Tests/TestHelpers.fs#L145) helper function since all your tests are like this.
See https://github.com/fsprojects/fantomas/blob/818c5a956be2fdaf6f4f756c0bd8e5277cd99f38/src/Fantomas.Tests/RecordTests.fs#L205

Reply:
Yup makes sense - pushed a change